great_expectations/experimental/datasources/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -247,7 +252,30 @@ def _generate_batch_spec_parameters_from_batch_definition(
 
         path = self._get_full_file_path(path=path)
 
-        return {"path": path}
+        return {FilePathDataConnector.FILE_PATH_BATCH_SPEC_KEY: path}
+
+    def _insure_regex_groups_include_data_reference_key(
+        self, regex: re.Pattern
+    ) -> re.Pattern:
+        """
+        Args:
+            regex: regex pattern for filtering data references; if reserved group name "path" (FILE_PATH_BATCH_SPEC_KEY)
+            is absent, then it is added to enclose original regex pattern, supplied on input.
+
+        Returns:
+            Potentially modified Regular Expression pattern (with enclosing FILE_PATH_BATCH_SPEC_KEY reserved group)
+        """
+        regex_parser: RegExParser = RegExParser(
```

Comment:
Did you need to put this type annotation here due to the type checker? Otherwise I would remove it.

Reply:
@NathanFarmer No, my old habit (deleting -- in next commit).  Thanks.